### PR TITLE
393 definition of regelaltersgrenze not correct

### PR DESCRIPTION
**Pull request by TeBackh**
### What problem do you want to solve?

Problem of Issue 393. I fixed the parameters of the piecewise function for the regelaltersgrenze.

### Todo

- [X] Replace cohort thresholds and adjustment rates for 2007 reform
- [X] If it is a relevant contribution, put it in `CHANGES.rst`.
- [ ] check rules for cohorts before 1945.
- [ ] create tests

### CODE REVIEW

File: gettsim/parameters/ges_rente.yaml [hunk 1]
```diff
@@ -730,30 +730,84 @@ regelaltersgrenze:
...
+      Diese Funktion ist hier ab 2002 angegeben. Vor 2002 ist das Steuersystem nicht
+      implementiert. Kohorten vor 1937 konnten mit 63 abschlagsfrei in Rente gehen.
+      Ab Geburtsdatum Januar 1937 wurde fuer jeden Monat, den die Person spaeter geboren wurde das
+      Eintrittsalter um 1 Monat erhoeht bis Geburtsdatum Dezember 1938: Eintrittsalter 65. Dieses
+      Eintrittsalter galt bis einschliesslich Jahrgang 1948. Ab Geburtsdatum Maerz 1949 entspricht die
+      Altersgrenze der Regelaltersgrenze, also 65 Jahre, 3 Monate. Fuer Geburtsdatum 1/1949 ist sie
+      65, 1 Monat. Fuer 2/1949 65, 2 Monate.
+    en:
+  type: piecewise_linear
+  2002-01-01:
```
Comment (hmgaudecker) [inline gettsim/parameters/ges_rente.yaml:788]: Again, use earliest date where this makes sense. Might need to distinguish men and women?

File: gettsim/parameters/ges_rente.yaml [hunk 2]
```diff
@@ -730,30 +730,84 @@ regelaltersgrenze:
...
+      upper_threshold: inf
+      rate_linear: 0
+
+altersgrenze_langjahrig:
+  name:
+    de: Eintrittsalter für langjaehrig versicherte - abschlagsfrei
+    en: Entry age for long term insured (35 years) without deductions
+  description:
+    de:
+      Stückweise lineare Funktion, die die Regelaltersgrenze angibt, bei der das Individuum mit seinem vollen Rentenanspruch in Rente
```
Comment (hmgaudecker) [inline gettsim/parameters/ges_rente.yaml:769]:
```suggestion
      Stückweise lineare Funktion, die die Regelaltersgrenze angibt, bei der jemand mit mindestens 35 Beitragsjahren mit  vollem Rentenanspruch in Rente
```
Add qualifier everywhere, very easy to get confused otherwise.

File: gettsim/tests/test_data/grundrente_proxy_rente.csv [hunk 3]
```diff
@@ -1,9 +1,9 @@
 "p_id","tu_id","hh_id","jahr","entgeltp","wohnort_ost","priv_rente_m","jahr_renteneintr","alter","bruttolohn_m","geburtsjahr","rentner","rente_vorj_vor_grundr_proxy_m","note"
-1,1,1,2021,20,"False","0","2016",70,0,1951,"True","675.59","made up, regression test"
-2,2,2,2021,30,"False","0","2016",70,0,1951,"True","1013.39","made up, regression test"
-3,3,3,2021,20,"True","0","2016",70,0,1951,"True","656.62","made up, regression test"
-4,4,4,2021,20,"False","1000","2016",70,0,1951,"True","1675.59","made up, regression test"
-5,5,5,2021,20,"False","0","2014",70,0,1951,"True","626.36","made up, regression test"
-6,6,6,2021,20,"False","0","2020",70,0,1951,"True","834.24","made up, regression test"
-7,7,7,2021,20,"False","0","1996",90,0,1931,"True","683.8","made up, regression test"
+1,1,1,2021,20,"False","0","2016",70,0,1951,"True","673.54","made up, regression test"
```
Comment (hmgaudecker) [inline gettsim/tests/test_data/grundrente_proxy_rente.csv:2]: Please add test cases for the edge cases, e.g., year 2000 and age 63. I do not see through how this is interacting with `regelaltersgrenze`.

**Review by hmgaudecker — COMMENTED** [review]
Great stuff, congratulations! 

I think this will need some clarifications and it would be great to add tests for this. Essentially all of that are inherited problems, only partly related to the actual stuff added here.

- e.g., test whether `regelaltersgrenze` caught correctly for various cohorts 
- similar for `altersgrenze_langjährig_versicherte`, e.g., year 2000 and age 63. I do not see through how this is interacting with `regelaltersgrenze` in the implementation.

File: CHANGES.rst [hunk 4]
```diff
@@ -12,6 +12,8 @@ releases are available on `Anaconda.org <https://anaconda.org/gettsim/gettsim>`_
   (:ghuser:`Eric-Sommer`, :ghuser:`ChristianZimpelmann`).
 * :gh:`385` Make altersentlastungsbetrag dependent on age not on current
   date (:ghuser:`m-pannier`, :ghuser:`lillyfischer`).
+* :gh:`393` normal retirement age adjustment aligned with the rules.
```
[the remark this answers is not in the document]
Comment (TeBackh) [reply CHANGES.rst:15]: for the long term contributors I just added the parameters. It is not implemented yet. (bad practice I guess - sorry for that) Therefore, I would only aggree to the first part of the sentence.

File: gettsim/tests/test_data/grundrente_proxy_rente.csv (hunk 3 above)
Comment (TeBackh) [reply gettsim/tests/test_data/grundrente_proxy_rente.csv:2]: The change in regelaltersgrenze affects the deductions of benefits, therefore the made-up pension payments (by the previous gettsim implementation) used in these test cases. In the fixed statutory retirement age is 1 month higher now for the cases , therefore penalties are higher and pensions lower. But yes, it seems like we need better tests for the implementation of the regular pension payments.

File: gettsim/parameters/ges_rente.yaml [hunk 5]
```diff
@@ -730,30 +730,84 @@ regelaltersgrenze:
...
+      wenn keine Ausnahmeregelungen erfüllt sind.
     en:
       Piecewise linear function returning the regelaltersgrenze at which the agent is allowed to get pensions with his full
       claim. If the agent retires earlier or later, the Zugangsfaktor and therefore
-      the pension claim is higher or lower.
+      the pension claim is higher or lower, unless special conditions apply.
   reference: § 35 Satz 2 SGB VI
   note:
     de: >-
       Diese Funktion ist hier ab 2002 angegeben. Vor 2002 ist das Steuersystem nicht
```
[the remark this answers is not in the document]
Comment (TeBackh) [reply gettsim/parameters/ges_rente.yaml:742]: The parameters here should reflect the normal retirement age 65 for cohorts before 1947 and for later cohorts adapted according to the reform. 
To my knowledge the "Regelaltersgrenze" as such is not different for women. 
There used to be the "pension for women" (up to cohort 1951) though, which is not implemented yet. (As many other pathways to retirement without deductions "Rente für langjährig Versicherte", "Rente für besonders langjährig Versicherte". ) I would prefer treating women of the affected cohorts as special cases, since after 1952 the "pension for women" was abolished.

File: CHANGES.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (TeBackh) [reply CHANGES.rst:17]: can be deleted. I added it now to my new pull request and will add a note there.

File: gettsim/parameters/ges_rente.yaml (hunk 1 above)
Comment (TeBackh) [reply gettsim/parameters/ges_rente.yaml]: is (re)moved now to another issue

File: gettsim/parameters/ges_rente.yaml (hunk 2 above)
Comment (TeBackh) [reply gettsim/parameters/ges_rente.yaml]: moved to another issue

**Comment by ChristianZimpelmann** [comment]
It seems like all comments are addressed now except the request for more tests.

I think we should add `ges_rente_regelaltersgrenze` to the tests at `test_renten_anspr` and potentially add some more test cases there (esp. the edge cases).

@TeBackh, would you like to do this in this PR? Otherwise, @LauraGergeleit could put it on her ToDo-list, but probably only after the exams. I would then suggest to merge this PR without the additional test cases and open an issue.

**Comment by TeBackh** [comment]
> It seems like all comments are addressed now except the request for more tests.
> 
> I think we should add `ges_rente_regelaltersgrenze` to the tests at `test_renten_anspr` and potentially add some more test cases there (esp. the edge cases).
> 
> @TeBackh, would you like to do this in this PR? Otherwise, @LauraGergeleit could put it on her ToDo-list, but probably only after the exams. I would then suggest to merge this PR without the additional test cases and open an issue.

I think it would be good to do the tests this outside of this PR (perhaps when the other extensions of the pension system (PR #396) are established).

File: CHANGES.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (TeBackh) [reply CHANGES.rst:17]: is deleted
